data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart

```diff
@@ -161,4 +162,68 @@ class SharedSpaceDocumentDataSourceImpl implements SharedSpaceDocumentDataSource
 
     return taskIds.map((taskId) => DownloadTaskId(taskId));
   }
+
+  @override
+  Future<Uri> downloadNodeIOS(WorkGroupNode workgroupNode, Token token, Uri baseUrl, CancelToken cancelToken) async {
+    final streamController = StreamController<Uri>();
+    try {
+      await Future.wait([
+        _linShareHttpClient.downloadDocumentIOS(
+            Endpoint.sharedSpaces
+                .withPathParameter(workgroupNode.sharedSpaceId.uuid)
+                .withPathParameter('nodes')
+                .withPathParameter(workgroupNode.workGroupNodeId.uuid)
+                .withPathParameter(Endpoint.download)
+                .generateEndpointPath(),
+            cancelToken,
+            token),
+        getTemporaryDirectory()
+      ]).then((values) {
+        final fileStream = (values[0] as ResponseBody).stream;
```

Comment:
WDYT about testing if values is not empty ?

Reply:
Oh I don't think we need that because we already have `try catch` for that.